File: typescript/easy/counting-bits.ts
function countBits(n: number): number[] {
  let res: number[] = []

  for (let i = 0; i <= n; i++) {
    const binary = i.toString(2)
    res.push(binary.split('').filter((x) => x === '1').length)
  }

  return res;
};

File: go/counting-bits.go
package main

import "strconv"

func countBits(n int) []int {
	res := []int{}

	for i := 0; i <= n; i++ {
		binary := strconv.FormatInt(int64(i), 2)

		count := 0
		for _, v := range binary {
			if v == '1' {
				count++
			}
		}
		res = append(res, count)
	}

	return res
}
